commit/merkleroot/rmn/client.go

```diff
@@ -541,12 +541,56 @@ func (c *client) listenForRmnReportSignatures(
 			reportSig := responseTyp.GetReportSignature()
 			if reportSig == nil {
 				c.lggr.Infof("RMN node returned an unexpected type of response: %+v", responseTyp.Response)
-			} else {
-				c.lggr.Infow("received report signature", "node", resp.RMNNodeID, "requestID", responseTyp.RequestId)
-				reportSigs = append(reportSigs, reportSig)
+				continue
 			}
 
-			if len(reportSigs) >= c.minSigners {
+			rmnNode, exists := c.getRmnNodeByID(resp.RMNNodeID)
+			if !exists {
+				c.lggr.Warnw("rmn node that appears in report signature does not exist", "nodeID", resp.RMNNodeID)
+				continue
+			}
+
+			ch, exists := chainsel.ChainBySelector(destChain.DestChainSelector)
+			if !exists {
+				c.lggr.Warnw("unknown chain selector", "chainSelector", destChain.DestChainSelector)
+				continue
+			}
+
+			laneUpdates, err := NewLaneUpdatesFromPB(fixedDestLaneUpdates)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert lane updates from protobuf: %w", err)
+			}
+
+			rmnReport := cciptypes.RMNReport{
+				ReportVersion:               c.rmnCfg.Home.RmnReportVersion,
+				DestChainID:                 cciptypes.NewBigIntFromInt64(int64(ch.EvmChainID)),
+				DestChainSelector:           cciptypes.ChainSelector(destChain.DestChainSelector),
+				RmnRemoteContractAddress:    c.rmnCfg.Remote.ContractAddress,
+				OfframpAddress:              destChain.OfframpAddress,
+				RmnHomeContractConfigDigest: c.rmnCfg.Home.ConfigDigest,
+				LaneUpdates:                 laneUpdates,
+			}
+
+			sig, err := NewECDSASigFromPB(reportSig.Signature)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert signature from protobuf: %w", err)
+			}
+
+			err = c.rmnCrypto.VerifyReportSignatures(
```

Comment:
Is there a reason we verify report signatures twice? This code runs in query (I think) and we seem to also do it in observation

Reply:
Because we might have a malicious leader that sends wrong sigs.
cc @gtklocker correct me if i'm wrong